File: fix-all-broken-pages.js
const fs = require('fs');
const path = require('path');

function fixPageFile(filePa, t, h) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Fix missing anchor tags in CTA section
    const ctaPattern = /<div className='flex flex-col sm:flex-row gap-4 justify-center'>\s*\n\s*Start Your Project\s*\n\s*<\/a>\s*\n\s*Learn More\s*\n\s*<\/a>/gs;
    const ctaReplacement = `<div className='flex flex-col sm:flex-row gap-4 justify-center'>
              <a
                href='/contact' className='bg-gradient-to-r from-cyan-500 to-purple-600 text-white px-8 py-3 rounded-lg font-semibold hover:from-cyan-600 hover:to-purple-700 transition-all duration-300'>
                Start Your Project
              </a>
              <a
                href='/services' className='border border-white/20 text-white px-8 py-3 rounded-lg font-semibold hover:bg-white/10 transition-all duration-300'>
                Learn More
              </a>`;
    
    if (ctaPattern.test(conte, n, t)) {
      content = content.replace(ctaPattern, ctaReplacement);
      modified = true;
    }
    
    // Fix broken features map patterns
    const featuresPattern = /<div className='grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-8'>\s*<div className='w-16 h-16 bg-gradient-to-r from-cyan-500 to-purple-600 rounded-full flex items-center justify-center mx-auto mb-4'>\s*<feature\.icon className='w-8 h-8 text-white' \/>\s*<\/div>\s*<h3 className='text-xl font-semibold text-white mb-3'>\s*{feature\.title}\s*<\/h3>\s*<p className='text-gray-300'>{feature\.description}<\/p>\s*<\/div>\s*\)\)}/gs;
    const featuresReplacement = `{features.map((feature, index) => (
                <div
                  key={ ind, e, x }
                  className='bg-white/5 rounded-2xl p-8 backdrop-blur-lg border border-white/10 text-center'
                >
                  <div className='w-16 h-16 bg-gradient-to-r from-cyan-500 to-purple-600 rounded-full flex items-center justify-center mx-auto mb-4'>
                    <feature.icon className='w-8 h-8 text-white' />
                  </div>
                  <h3 className='text-xl font-semibold text-white mb-3'>
                    {feature.title}
                  </h3>
                  <p className='text-gray-300'>{feature.description}</p>
                </div>
              ))}`;
    
    if (featuresPattern.test(conte, n, t)) {
      content = content.replace(featuresPattern, featuresReplacement);
      modified = true;
    }
    
    if (modifi, e, d) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed: ${ filePa, t, h }`);
      return true;
    }
    
    return false;
  } catch (err, o, r) {
    console.error(`Error fixing ${ filePa, t, h }:`, error.message);
    return false;
  }
}

function findAndFixPages(d, i, r) {
  const files = fs.readdirSync(d, i, r);
  let fixedCount = 0;
  
  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePa, t, h);
    
    if (stat.isDirectory()) {
      fixedCount += findAndFixPages(filePa, t, h);
    } else if (file === 'page.tsx') {
      if (fixPageFile(filePa, t, h)) {
        fixedCount++;
      }
    }
  }
  
  return fixedCount;
}

const appDir = path.join(__dirname, 'app');
const fixedCount = findAndFixPages(appD, i, r);
console.log(`Fixed ${ fixedCou, n, t } page files`);

File: fix_all_broken_pages.py
#!/usr/bin/env python3
"""
Script to fix all broken page files by finding and replacing broken patterns
"""

import os
import glob
import re

def fix_broken_page(file_path):
    """Fix a broken page file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has broken patterns
        if 'return (' in content and '<>' in content and 'title>' in content:
            # Extract page name from file path
            page_name = os.path.basename(os.path.dirname(file_path)).replace('-', ' ').title().replace(' ', '')
            if page_name.endswith('Page'):
                page_name = page_name[:-4]
            
            # Create a proper page
            proper_content = f'''import React from 'react';
import {{ Helmet }} from 'react-helmet-async';
import {{ Link }} from 'react-router-dom';
import {{ ArrowRight }} from 'lucide-react';

export default function {page_name}Page() {{
  return (
    <>
      <Helmet>
        <title>{page_name} - Zion Tech Group</title>
        <meta name="description" content="Professional {page_name.lower()} solutions and services." />
      </Helmet>
      
      <div className="min-h-screen bg-gradient-to-br from-blue-900 via-purple-900 to-indigo-900 pt-20">
        <div className="max-w-4xl mx-auto px-4 sm:px-6 lg:px-8 py-12">
          <div className="text-center">
            <h1 className="text-4xl font-bold text-white mb-6">{page_name}</h1>
            <p className="text-lg text-gray-300 mb-8 max-w-2xl mx-auto">
              Professional {page_name.lower()} solutions and services.
            </p>
            <div className="flex flex-col sm:flex-row gap-4 justify-center">
              <Link
                to="/contact"
                className="inline-flex items-center px-6 py-3 bg-blue-600 text-white font-semibold rounded-lg hover:bg-blue-700 transition-colors"
              >
                Contact Us
                <ArrowRight className="ml-2 h-5 w-5" />
              </Link>
              <Link
                to="/ai-services"
                className="inline-flex items-center px-6 py-3 bg-transparent border-2 border-white text-white font-semibold rounded-lg hover:bg-white hover:text-gray-900 transition-colors"
              >
                Explore Services
              </Link>
            </div>
          </div>
        </div>
      </div>
    </>
  );
}}'''
            
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(proper_content)
            return True
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    """Fix all broken page files"""
    # Find all page.tsx files
    page_files = glob.glob('app/**/page.tsx', recursive=True)
    
    fixed_count = 0
    for page_file in page_files:
        if fix_broken_page(page_file):
            print(f"Fixed: {page_file}")
            fixed_count += 1
    
    print(f"Fixed {fixed_count} broken pages!")

if __name__ == "__main__":
    main()